### PR TITLE
Update the WinForms matching game from .NET 5 to .NET 8

### PR DESCRIPTION
**Pull request by adegeo**

### CODE REVIEW

File: windowsforms/matching-game/net8-windows/cs/readme.md [hunk 1]
```diff
@@ -14,8 +16,6 @@ This sample demonstrates a simple memory game where you try to match tiles in th
 
 ## Prerequisites
 
-- [Visual Studio 2019 version 16.8](https://visualstudio.microsoft.com/downloads/?utm_medium=microsoft&utm_source=docs.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2019+desktopguide+winforms)
+- [Visual Studio 2022 version 17.8](https://visualstudio.microsoft.com/downloads/?utm_medium=microsoft&utm_source=learn.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2022+desktopguide+winforms+migration)
 
-  - Enable the [Visual Studio Desktop workload](https://docs.microsoft.com/visualstudio/install/modify-visual-studio?view=vs-2019&preserve-view=true#modify-workloads).
-
-  - Enable the [.NET 5 individual component](https://docs.microsoft.com/visualstudio/install/modify-visual-studio?view=vs-2019&preserve-view=true#modify-individual-components).
+  - Enable the [Visual Studio Desktop workload](https://docs.microsoft.com/visualstudio/install/modify-visual-studio?view=vs-2022&preserve-view=true#modify-workloads).
```
Comment (gewarren) [inline windowsforms/matching-game/net8-windows/cs/readme.md:21]:
```suggestion
  - Enable the [Visual Studio Desktop workload](https://learn.microsoft.com/visualstudio/install/modify-visual-studio?view=vs-2022&preserve-view=true#modify-workloads).
```

File: windowsforms/matching-game/net8-windows/vb/MatchingGame/My Project/Resources.resx [hunk 2]
```diff
@@ -46,7 +46,7 @@
     
     mimetype: application/x-microsoft.net.object.binary.base64
     value   : The object must be serialized with 
-            : System.Runtime.Serialization.Formatters.Binary.BinaryFormatter
+            : System.Serialization.Formatters.Binary.BinaryFormatter
```
Comment (gewarren) [inline windowsforms/matching-game/net8-windows/vb/MatchingGame/My Project/Resources.resx:49]: Previous namespace was correct: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.serialization.formatters.binary.binaryformatter?view=net-8.0

File: windowsforms/matching-game/net8-windows/cs/MatchingGame/App.config [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+    <configSections>
+    </configSections>
+    <startup> 
+        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.8"/>
```
Comment (Tanya-Solyanik) [inline windowsforms/matching-game/net8-windows/cs/MatchingGame/App.config:6]: Why was this file added under the `net8-windows` configuration?

Comment (adegeo) [reply windowsforms/matching-game/net8-windows/cs/MatchingGame/App.config:6]: @Tanya-Solyanik This was carried over from using the VS Upgrade plugin. I guess I can delete it. Thanks for spotting that!

File: windowsforms/matching-game/net8-windows/vb/MatchingGame/My Project/Resources.resx (hunk 2 above)
Comment (adegeo) [reply windowsforms/matching-game/net8-windows/vb/MatchingGame/My Project/Resources.resx:49]: This file is automatically generated and because I used the VS Update tool to port the code from net45 to net8, it looks like it kept the "old" wording in that comment.

File: windowsforms/matching-game/net8-windows/vb/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+---
+# Matching game
+
+This sample demonstrates a simple memory game where you try to match tiles in the fewest amount of moves possible. This sample was migrated from the [.NET Framework version](https://github.com/dotnet/samples/tree/main/windowsforms/matching-game/net45/vb).
+
+## Prerequisites
+
+- [Visual Studio 2022 version 17.8](https://visualstudio.microsoft.com/downloads/?utm_medium=microsoft&utm_source=learn.microsoft.com&utm_campaign=inline+link&utm_content=download+vs2022+desktopguide+winforms+migration)
+
+  - Enable the [Visual Studio Desktop workload](https://docs.microsoft.com/visualstudio/install/modify-visual-studio?view=vs-2022&preserve-view=true#modify-workloads).
```
Comment (adegeo) [inline windowsforms/matching-game/net8-windows/vb/readme.md:21]:
```suggestion
  - Enable the [Visual Studio Desktop workload](https://learn.microsoft.com/visualstudio/install/modify-visual-studio?view=vs-2022&preserve-view=true#modify-workloads).
```

**Review by Tanya-Solyanik — APPROVED** [review]
